src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseAsyncMethodInAsyncContextTests.cs

```diff
@@ -0,0 +1,1258 @@
+                                                 Dim t As Task(Of Integer) = Nothing
+                                                 Dim result As Integer = {|#0:t.Result|}
+                                                 Return Task.FromResult(result)
+                                             End Function
+    End Sub
+End Module
+";
+            await CreateVBTestAndRunAsync(testVB, VerifyVB.Diagnostic(UseAsyncMethodInAsyncContext.DescriptorNoAlternativeMethod).WithLocation(0).WithArguments("Result"));
+        }
+
+        [Fact]
+        public async Task TaskOfTResultInTaskReturningMethodAnonymousDelegate_GeneratesNoWarning()
+        {
+            var testCS = @"
+using System;
+using System.Threading.Tasks;
+
+class Test {
+    Task<int> T() {
+        Task<int> task = null;
+        task.ContinueWith(t => { Console.WriteLine(t.Result); });
+        return Task.FromResult(1);
+    }
+}
+";
+            await CreateCSTestAndRunAsync(testCS);
+
+            var testVB = @"
+Imports System
+Imports System.Threading.Tasks
+
+Module Program
+    Sub Main()
+        Test()
+    End Sub
+    Function Test() As Task(Of Integer)
+        Dim task As Task(Of Integer) = Nothing
+        task.ContinueWith(Sub(t) Console.WriteLine(t.Result))
+        Return task.FromResult(1)
+    End Function
+End Module
+";
+            await CreateVBTestAndRunAsync(testVB);
+        }
+
+        [Fact]
+        public async Task TaskGetAwaiterGetResultInTaskReturningMethodGeneratesWarning()
+        {
+            var testCS = @"
+using System.Threading.Tasks;
+
+class Test {
+    Task T() {
+        Task t = null;
+        {|#0:t.GetAwaiter().GetResult()|};
+        return Task.FromResult(1);
+    }
+}
+";
+            await CreateCSTestAndRunAsync(testCS, VerifyCS.Diagnostic(UseAsyncMethodInAsyncContext.DescriptorNoAlternativeMethod).WithLocation(0).WithArguments("GetResult"));
+
+            var testVB = @"
+Imports System.Threading.Tasks
+Module Program
+    Sub Main()
+        Test()
+    End Sub
+    Function Test() As Task
+        Dim t As Task = Nothing
+        {|#0:t.GetAwaiter().GetResult()|}
+        Return task.FromResult(1)
+    End Function
+End Module
+";
+            await CreateVBTestAndRunAsync(testVB, VerifyVB.Diagnostic(UseAsyncMethodInAsyncContext.DescriptorNoAlternativeMethod).WithLocation(0).WithArguments("GetResult"));
+        }
+
+        [Fact]
+        public async Task SyncInvocationWhereAsyncOptionExistsInSameTypeGeneratesWarning()
+        {
+            var testCS = @"
+using System.Threading.Tasks;
+class Test {
+    Task T() {
+        {|#0:Foo(10, 15)|};
+        return Task.FromResult(1);
+    }
+    internal static void Foo(int x, int y) { }
+    internal static Task FooAsync(int x, int y) => null;
+}
+";
+            await CreateCSTestAndRunAsync(testCS, VerifyCS.Diagnostic(UseAsyncMethodInAsyncContext.Descriptor).WithLocation(0).WithArguments("Foo(10, 15)", "FooAsync"));
```

Comment:
This is what I was suggesting to test in a previous comment 😄 
I think the message feels weird with the arguments being there, esp that they can get too long and also the other argument is just a method name.

Reply:
I did not see your previous comment, though agree with you, now after using `symbol.ToDisplayString()` it would look better

Reply:
@buyaa-n If I recall correctly, I think @mavasani was preferring `symbol.Name`, but I'm not sure, so will let him confirm

Reply:
We had some discussion in PCA messaging and `symbol.ToDisplayString()` is what we end up using